File: test/languages/Sample.java
package fixtures;

import java.time.Instant;
import java.util.List;
import java.util.Optional;

public final class Sample {
    private static final int DEFAULT_LIMIT = 10;

    public enum State { DRAFT, PUBLISHED }
    public record Review(long id, String title, Optional<Double> score, State state) {}
    public sealed interface Result<T> permits Success, Failure {}
    public record Success<T>(T value) implements Result<T> {}
    public record Failure<T>(String message) implements Result<T> {}

    public static List<String> summarize(List<Review> reviews, int limit) {
        return reviews.stream()
            .filter(review -> review.state() == State.PUBLISHED)
            .sorted((left, right) -> Double.compare(
                right.score().orElse(0.0), left.score().orElse(0.0)))
            .limit(limit)
            .map(review -> "%s: %.1f".formatted(review.title(), review.score().orElse(0.0)))
            .toList();
    }

    public static void main(String[] args) {
        var review = new Review(1L, "Night Walk", Optional.of(8.5), State.PUBLISHED);
        System.out.println(summarize(List.of(review), DEFAULT_LIMIT));
    }
}

File: test/languages/sample.kt
package fixtures

enum class ReviewState { DRAFT, PUBLISHED }

@JvmInline
value class ReviewId(val value: Long)

data class Review(
    val id: ReviewId,
    val title: String,
    val score: Double? = null,
    val state: ReviewState = ReviewState.DRAFT,
)

sealed interface Result<out T> {
    data class Success<T>(val value: T) : Result<T>
    data class Failure(val message: String) : Result<Nothing>
}

fun List<Review>.summaries(limit: Int = 10): List<String> =
    asSequence()
        .filter { it.state == ReviewState.PUBLISHED }
        .sortedByDescending { it.score ?: 0.0 }
        .take(limit)
        .map { review -> "${review.title}: ${review.score?.let { "%.1f".format(it) } ?: "unrated"}" }
        .toList()

fun main() = println(listOf(Review(ReviewId(1), "Night Walk", 8.5)).summaries())
